File: src/main/java/site/gachontable/gachontablebe/domain/menu/domain/Menu.java
package site.gachontable.gachontablebe.domain.menu.domain;

import jakarta.persistence.*;
import lombok.*;
import site.gachontable.gachontablebe.domain.pub.domain.Pub;

@Entity(name = "menu")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Menu {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer menuId;

    @ManyToOne(fetch = FetchType.LAZY)
    private Pub pub;

    @Column(nullable = false)
    private String menuName;

    @Column(nullable = false)
    private String price;

    @Column(nullable = false)
    private String oneLiner;

    @Column
    private String thumbnail;

    public static Menu create(Pub pub, String menuName, String price, String oneLiner, String thumbnail) {
        return Menu.builder()
                .pub(pub)
                .menuName(menuName)
                .price(price)
                .oneLiner(oneLiner)
                .thumbnail(thumbnail)
                .build();
    }

    @Builder
    public Menu(Pub pub, String menuName, String price, String oneLiner, String thumbnail) {
        this.pub = pub;
        this.menuName = menuName;
        this.price = price;
        this.oneLiner = oneLiner;
        this.thumbnail = thumbnail;
    }

    public void update(String menuName, String price, String oneLiner, String thumbnail) {
        this.thumbnail = thumbnail;
        this.menuName = menuName;
        this.price = price;
        this.oneLiner = oneLiner;
    }
}

File: src/main/kotlin/site/gachontable/domain/menu/domain/Menu.kt
package site.gachontable.domain.menu.domain

import jakarta.persistence.*
import site.gachontable.domain.pub.domain.Pub

@Entity(name = "menu")
class Menu(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = 0,

    @ManyToOne(fetch = FetchType.LAZY)
    val pub: Pub,

    @Column(nullable = false)
    var menuName: String,

    @Column(nullable = false)
    var price: String,

    @Column(nullable = false)
    var oneLiner: String,

    @Column
    var thumbnail: String,
) {
    fun update(menuName: String, price: String, oneLiner: String, thumbnail: String) {
        this.thumbnail = thumbnail
        this.menuName = menuName
        this.price = price
        this.oneLiner = oneLiner
    }

    companion object {
        fun create(
            pub: Pub, menuName: String, price: String, oneLiner: String, thumbnail: String,
        ): Menu {
            return Menu(
                pub = pub,
                menuName = menuName,
                price = price,
                oneLiner = oneLiner,
                thumbnail = thumbnail,
            )
        }
    }
}
